staging/src/k8s.io/apiserver/pkg/endpoints/discovery/etag.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package discovery
+
+import (
+	"crypto/sha512"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// This file exposes helper functions used for calculating the E-Tag header
+// used in discovery endpoint responses
+
+func CalculateETag(resurces *metav1.APIResourceList) (string, error) {
```

Comment:
What I really want to do is just remove `APIResourceLister` interface and change the APIVersionHandler to take `[]metav1.APIResource` directly in its constructor. Not sure if I have discretion to do this. That would allow CalculcateEtag to be private. It currently needs to be exported to be used here: https://github.com/Jefftree/kubernetes/pull/5/files#diff-68dc09ed8ce13daada37fb11dc0a7433fb22576e8d66c89b615f5e6a8f55218aR192